Framework/Algorithms/src/Regroup.cpp

```diff
@@ -65,15 +63,9 @@ void Regroup::exec() {
   // Get the input workspace
   MatrixWorkspace_const_sptr inputW = getProperty("InputWorkspace");
 
-  // can work only if all histograms have the same boundaries
-  if (!API::WorkspaceHelpers::commonBoundaries(*inputW)) {
-    g_log.error("Histograms with different boundaries");
-    throw std::runtime_error("Histograms with different boundaries");
-  }
```

Comment:
Is it safe to remove this and checks like it in other algorithms?

Reply:
I tried to very carefully check that in these cases the validation can be/is done by `CommonBinsValidator` and added the validator where needed.

These checks seem to be an ancient remnant of times when `CommonBinsValidator` did not check the binning properly and the algorithms needed to do more throughout checking in `exec()`.

Reply:
This is a correct assumption as the validator used to only check first and last spectra I think.